pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java

```diff
@@ -172,15 +180,22 @@ private boolean isNewSegmentMetadata(@Nonnull OfflineSegmentZKMetadata newSegmen
   }
 
   @Nonnull
-  private String downloadSegmentToLocal(@Nonnull String uri, @Nonnull String tableName, @Nonnull String segmentName)
+  private String downloadSegmentToLocal(@Nonnull String uri, PinotCrypter crypter, @Nonnull String tableName, @Nonnull String segmentName)
       throws Exception {
     File tempDir = new File(new File(_instanceDataManager.getSegmentFileDirectory(), tableName),
         "tmp_" + segmentName + "_" + System.nanoTime());
     FileUtils.forceMkdir(tempDir);
-    File tempTarFile = new File(tempDir, segmentName + ".tar.gz");
+    File tempDownloadFile = new File(tempDir, segmentName + ENCODED_SUFFIX);
+    File tempTarFile = new File(tempDir, segmentName + TAR_GZ_SUFFIX);
     File tempSegmentDir = new File(tempDir, segmentName);
     try {
-      SegmentFetcherFactory.getInstance().getSegmentFetcherBasedOnURI(uri).fetchSegmentToLocal(uri, tempTarFile);
+      SegmentFetcherFactory.getInstance().getSegmentFetcherBasedOnURI(uri).fetchSegmentToLocal(uri, tempDownloadFile);
+      if (crypter != null) {
+        crypter.decrypt(tempDownloadFile, tempTarFile);
```

Comment:
might not always be .tar.gz right?

Reply:
How does that work today, ie before this PR?

Reply:
Segments were always in .tar.gz format

Reply:
I am probably missing something, segment file will either be a tar.gz or tar.gz + encrypted. Is there another format that is currently supported?

Reply:
Discussed offline, this is not an issue as files are always in tar.gz format (and optionally encrypted on top of it).